src/Adapter/PDF/OrderReturnPdfGenerator.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\PDF;
+
+use Context;
+use PDF;
+use PrestaShop\PrestaShop\Adapter\OrderReturn\Repository\OrderReturnRepository;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\OrderReturnSettings;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\ValueObject\OrderReturnId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\PDF\PDFGeneratorInterface;
+use RuntimeException;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Validate;
+
+/**
+ * Generates invoice for given order.
+ */
+final class OrderReturnPdfGenerator implements PDFGeneratorInterface
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var OrderReturnRepository
+     */
+    private $orderReturnRepository;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param OrderReturnRepository $orderReturnRepository
+     */
+    public function __construct(TranslatorInterface $translator, OrderReturnRepository $orderReturnRepository)
+    {
+        $this->translator = $translator;
+        $this->orderReturnRepository = $orderReturnRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generatePDF(array $orderReturnId)
```

Comment:
Again, we should have a new interface instead of using the crappy legacy one that takes an array.

Reply:
I introduced new PR which would be start of wider PDF refactor.
https://github.com/PrestaShop/PrestaShop/pull/33027

But refactoring PDF is a lot of work and I don't think it should be done in this PR since it will add extra bloat and make it even harder to merge.